doc/tutorials/tutorial_understanding_building_benchmark.rst

```diff
@@ -0,0 +1,312 @@
+.. _Benchopt: https://benchopt.github.io/#
+
+Understanding and building a simple benchmark with Benchopt
+===========================================================
+
+Are you looking for a simple, step-by-step introduction to the `Benchopt`_ library? Then look no further! In this tutorial, we are going to learn how to use the `Benchopt`_ library to compare various algorithms in order to minimize a given cost. This tutorial complements the `write a benchmark <https://benchopt.github.io/how.html>`_ webpage with a more practical, hands-on approach. 
+
+Let's say we want to minimize the cost function of the `ridge regression problem <https://en.wikipedia.org/wiki/Ridge_regression>`_, a variant of linear regression which is a common of supervised machine learning.
+
+.. math::
+
+    f(\beta) = \frac{1}{2} {\|y - X\beta\|}^2 + \lambda{\|\beta\|}^2
+
+where
+    - \\(y\\) is a vector with \\(m\\) elements, typically some observations
+    - \\(X\\) is a matrix with \\(m \\times n\\) elements, typically the regressors
+    - :math:`\beta` is a vector with \\(n\\) elements that we obtain by regressing \\(y\\) on \\(X\\).
+    - \\(\\|x\\|\\) is the euclidean norm of \\(x\\).
+     
+
+Let us also suppose that we are working with a specific optimization algorithm that aims at solving the ridge regression problem: `the gradient descent (GD) algorithm <https://en.wikipedia.org/wiki/Gradient_descent>`_. GD is a simple algorithm that moves iteratively along the steepest descent direction. Given an estimate :math:`\beta`, GD iterates the following step:
+
+.. math:: 
+
+    \beta \leftarrow \beta - \eta \nabla_{f}(\beta)
+
+where :math:`\eta` is the learning rate or stepsize, and controls how much the unknown is moved alongside the descent direction. Vector :math:`\nabla_{f}(\beta)` is the gradient of the cost function defined above, evaluated at \\(\\beta\\).
+
+Even though GD is a simple algorithm, when applied to our ridge regression problem, we should be mindful of the two hyperparameters: the regularization parameter \\(\\lambda\\), and the gradient stepsize \\( \\eta\\). It is very common in machine learning that we are facing a situation where we want to vary these parameters, and compare the behavior of our algorithm in each scenario. Benchopt we can create a benchmark for this problem, here is a screenshot of what we will get at the end of this tutorial.
+
+.. image:: screenshot_mytemplate_ridgeloss.png
```

Comment:
GD and GD-ridge are the two solvers we get at the end of the tutorial. As stated, this is a screenshot from the future, I think it is fine if not everything is clear yet, since explaining this is the whole point of the tutorial.

The name of the Objective (OLS) is something I forgot to change in the tutorial.

By naming the benchmark my_benchmark, the plot prints My Benchmark benchmark as a title, which I agree is not great. Any suggestion?

Reply:
The `GD` solver is designed for OLS and we change the benchmark to tackle Ridge regression.
And we also want the tuto to explain how to add a solver. So the idea was to had a correct solver, vs the existing one that does not account for regularisation. This allow to show an "interesting" thing, that is that if you don't optimize the right problem, it does not work :)

But maybe we can simply modify the GD one directly and add a diverging one with step size too large? I have no strong opinion here.